File: tests/fixtures/codegen/dimensions/structs-expected/typescript/runtime/queries/arrays/GetChoice.ts
import type { GetChoiceParams, GetChoiceRow } from "../../../types/queries/arrays/GetChoice.js";
export type { GetChoiceParams, GetChoiceRow } from "../../../types/queries/arrays/GetChoice.js";
import type * as pgsid from "./pgsid/queryable.js";
export const getChoiceSql = "SELECT CASE WHEN id > 0 THEN matrix ELSE ordinary END AS mixed FROM arrays;";
export async function getChoice(db: pgsid.Queryable): Promise<GetChoiceRow | undefined> {
    const result = await db.query(getChoiceSql, []);
    const row = result.rows[0];
    if (row === undefined)
        return undefined;
    return row as GetChoiceRow;
}

File: tests/fixtures/codegen/dimensions/pointers-expected/queries/arrays/getchoice.go
package arrays

import (
	"context"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
	"github.com/jackc/pgx/v5/pgtype"
)

const GetChoiceSQL = "SELECT CASE WHEN id > 0 THEN matrix ELSE ordinary END AS mixed FROM arrays;"

type GetChoiceParams struct {
}
type GetChoiceRow struct {
	Mixed pgtype.Array[int32] `db:"mixed"`
}

func (q *Queries) GetChoice(ctx context.Context) (GetChoiceRow, error) {
	var row GetChoiceRow
	err := q.db.QueryRow(ctx, GetChoiceSQL).Scan(pgsidpgx.ArrayTarget[int32](&row.Mixed, []int{1, 2}, "GetChoice", "mixed", false))
	return row, err
}
